File: backend/internal/model/comment.go
package model

import "time"

type Comment struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	Content   string    `gorm:"type:text;not null" json:"content"`
	UserID    uint      `gorm:"not null;index" json:"user_id"`
	User      *User     `gorm:"foreignKey:UserID" json:"user,omitempty"`
	ProjectID *uint     `gorm:"index" json:"project_id,omitempty"`
	TaskID    *uint     `gorm:"index" json:"task_id,omitempty"`
	ParentID  *uint     `gorm:"index" json:"parent_id,omitempty"`
	Parent    *Comment  `gorm:"foreignKey:ParentID" json:"parent,omitempty"`
	Replies   []Comment `gorm:"foreignKey:ParentID" json:"replies,omitempty"`
	CreatedAt time.Time `json:"created_at"`
}

File: frontend/src/api/comment.ts
import api from './index'

export interface Comment {
  id: number
  content: string
  user_id: number
  username: string
  avatar_url: string
  parent_id: number | null
  replies: Comment[]
  created_at: string
  updated_at: string
}

export interface CommentListResponse {
  list: Comment[]
  total: number
  page: number
  page_size: number
}

export function getProjectComments(projectId: number | string, params?: { page?: number; page_size?: number }) {
  return api.get<CommentListResponse>(`/api/projects/${projectId}/comments`, { params })
}

export function createProjectComment(projectId: number | string, data: { content: string }) {
  return api.post(`/api/projects/${projectId}/comments`, data)
}

export function getTaskComments(taskId: number | string) {
  return api.get<Comment[]>(`/api/tasks/${taskId}/comments`)
}

export function createTaskComment(taskId: number | string, data: { content: string }) {
  return api.post(`/api/tasks/${taskId}/comments`, data)
}

export function replyToComment(commentId: number | string, data: { content: string }) {
  return api.post(`/api/comments/${commentId}/reply`, data)
}

export function deleteComment(id: number | string) {
  return api.delete(`/api/comments/${id}`)
}
